build.py

```diff
@@ -287,14 +288,26 @@ def do_make(target=None, log=False):
 
 # Override info using environment variables
 if use_environment:
+    required_environment = ["COMMIT_ID", "TREE", "BRANCH", "GIT_DESCRIBE", "GIT_DESCRIBE_VERBOSE"]
```

Comment:
I think the things that can be derived from the git repo shouldn't be mandatory environment variables.

Reply:
They are only mandatory if you have told build.py to read from environment variables. In which case it doesnt try and use the git repo and all of these are needed.

Reply:
In some cases such as for the automated boot bisection (`bisect.jpl`), both `-e` and `-g` are used to intialise variables from the git repo and override some from the environment.

Reply:
As per our irc discussion, this has been removed